app/assets/javascripts/components/articles/article_graphs.jsx

```diff
@@ -1,170 +1,153 @@
-  },
-
-  graphId() {
-    return `vega-graph-${this.props.article.id}`;
-  },
-
-  render() {
-    let style = 'hidden';
-    if (this.state.showGraph) {
-      style = '';
+
+    if (dataIncludesWp10 && selectedRadio === 'wp10_score') {
+      return (
+        <Wp10Graph
+          graphid={graphId()}
+          graphWidth={500}
+          graphHeight={300}
+          articleData={articleData}
+        />
+      );
     }
 
-    let graph;
-    let editSize;
-    let radioInput;
-    const graphWidth = 500;
-    const graphHeight = 300;
-    const className = `vega-graph ${style}`;
-
-    if (this.state.articleData != null) {
-      // Only render the wp10 graph radio button if the data includes wp10 / article completeness scores
-      if (this.state.articleData[0].wp10) {
-        radioInput = (
-          <div>
-            <div className="input-row">
-              <input
-                type="radio"
-                name="wp10_score"
-                value="wp10_score"
-                checked={this.state.selectedRadio === 'wp10_score'}
-                onChange={this.handleRadioChange}
-              />
-              <label htmlFor="wp10_score">{I18n.t('articles.wp10')}</label>
-            </div>
-            <div className="input-row">
-              <input
-                type="radio"
-                name="edit_size"
-                value="edit_size"
-                checked={this.state.selectedRadio === 'edit_size'}
-                onChange={this.handleRadioChange}
-              />
-              <label htmlFor="edit_size">{I18n.t('articles.edit_size')}</label>
-            </div>
-          </div>
-        );
-        if (this.state.selectedRadio === 'wp10_score') {
-          graph = (
-            <Wp10Graph
-              graphid = {this.graphId()}
-              graphWidth = {graphWidth}
-              graphHeight = {graphHeight}
-              articleData = {this.state.articleData}
-            />
-          );
-        } else {
-          graph = (
-            <EditSizeGraph
-              graphid ={this.graphId()}
-              graphWidth = {graphWidth}
-              graphHeight = {graphHeight}
-              articleData = {this.state.articleData}
-            />
-          );
-        }
-      } else {
-        editSize = (
-          <p>{I18n.t('articles.edit_size')}</p>
-        );
-        graph = (
-          <EditSizeGraph
-            graphid ={this.graphId()}
-            graphWidth = {graphWidth}
-            graphHeight = {graphHeight}
-            articleData = {this.state.articleData}
-          />
-        );
-      } // Display the loading element if articleData is not available
-     } else {
-      graph = <Loading />;
+    return (
+      <EditSizeGraph
+        graphid={graphId()}
+        graphWidth={500}
+        graphHeight={300}
+        articleData={articleData}
+      />
+    );
+  };
+
+  const radioInput = () => {
+    if (!dataIncludesWp10 || !articleData) {
+      return null;
```

Comment:
similar to https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5498/files#r1365943358